controller/internal/routes/query_options.go

```diff
@@ -31,46 +31,22 @@ const (
 	OffsetDefault = 0
 )
 
-type QueryOptions struct {
+// Represents external query options, which limits some query options (i.e. return all vs max limit). PublicQueryOptions
+// can be used internally as long as the public option limitations are fitting.
+type PublicQueryOptions struct {
```

Comment:
Renamed so it is clear that this struct is used for public API responses (currently only the REST API)